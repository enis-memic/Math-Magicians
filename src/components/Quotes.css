.quote-container div {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}
p {
  margin: 1rem;
}
