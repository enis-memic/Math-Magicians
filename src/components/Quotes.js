import React, { useState, useEffect } from 'react';
import './Quotes.css';

const Quote = () => {
  //   const [quote, setQuote] = useState('');
  //   const [author, setAuthor] = useState('');
  const [post, setPost] = useState(null);
  const [loading, setLoading] = useState(true);
  const [error, setError] = useState(null);

  useEffect(() => {
    setLoading(true);
    fetch('https://api.api-ninjas.com/v1/quotes?category=learning', {
      headers: { 'X-Api-Key': 'iKNK7Xbs47dymQExJ6KWaQ==1WtvRayZcMIMkK2x' },
    })
      .then((response) => response.json())
      .then((data) => {
        const body = `${data[0].quote} -  by ${data[0].author}`;
        setPost(body);
        setLoading(false);
      })
      .catch(() => {
        setError('Error fetching.');
        setLoading(false);
      });
  }, []);

  return (
    <div className="quote-container">
      {loading && <p>Loading...</p>}
      {error && <p>{error}</p>}
      {!loading && !error && (
        <div>
          <p>{post}</p>
        </div>
      )}
    </div>
  );
};

export default Quote;
