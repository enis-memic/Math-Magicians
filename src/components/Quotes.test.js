import renderer from 'react-test-renderer';
import Quotes from './Quotes';

it('Qoute page is working', () => {
  const tree = renderer.create(<Quotes />).toJSON();
  expect(tree).toMatchSnapshot();
});
