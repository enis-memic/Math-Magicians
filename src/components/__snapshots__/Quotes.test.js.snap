// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Qoute page is working 1`] = `
<div>
  <p>
    Loading...
  </p>
</div>
`;
